Improve setup script with better error handling and feedback

IronJam11 opened the pull request:
## Summary
This PR enhances the project setup script with proper error handling, dependency checking, and user feedback. The improvements make the script more robust and user-friendly while fixing the incorrect virtual environment deactivation command.

## Changes Made
- Added error handling with `set -e`
- Created error and success message functions with color coding
- Added dependency checks for python3, git, and wget
- Fixed virtual environment deactivation (removed incorrect `source deactivate`)
- Added progress messages for better user feedback
- Improved variable quoting for better security
- Added comments for better maintainability

## Testing Done
- Tested script on a clean environment
- Verified error handling by simulating various failure conditions:
  - Missing dependencies
  - Network issues during downloads
  - Permission issues
- Confirmed proper virtual environment activation and deactivation
- Tested on bash 5.0+ environment

## Issues
Fixes #214 

## Checklist
- [x] Code follows project style guidelines
- [x] Added comments for better maintainability
- [x] Tested on a clean environment
- [x] Error handling for all critical operations
- [x] User feedback for all major steps

Aditya062003 commented:
Good work! But, iirc there were some issues about setup script failing in some scenarios, it would be great if you could check that, and resolves those here itself.

IronJam11 commented:
So turns out the following changes helped: 
- Increasing the git buffer size 
- Using HTTPS over HTTP/2
<img width="1083" alt="Screenshot 2025-02-18 at 2 08 46 AM" src="https://github.com/user-attachments/assets/0a0d471d-a493-4012-a326-060a6c2ab001" />

Aditya062003 commented:
Great! I noticed both of these PRs are addressing similar issues: (https://github.com/AOSSIE-Org/EduAid/pull/188). It would be best if both of you collaborated to integrate these fixes into a single PR. You can work together to refine the solution and submit a combined PR. Let me know if you need any help! 🚀

IronJam11 commented:
So basically we have made a few improvments after discussing it together :- 
- Added debugging 
- Resolved the problem with installing the thinc and torch packages 
- Added documentation 
- Removed deactivate command so that the user can have the server running on script execution (tested) 
- Added colours for representing errors and successes correspondingly 
- Add command in script to perform the setup automatically (extraction,running server)
